packages/providers/src/coin-quantity.ts

```diff
@@ -24,9 +24,10 @@ export const coinQuantityfy = (coinQuantityLike: CoinQuantityLike): CoinQuantity
     max = coinQuantityLike.max ?? undefined;
   }
 
+  const bnAmount = bn(amount);
   return {
     assetId: hexlify(assetId),
-    amount: bn(amount),
+    amount: bnAmount.toNumber() > 0 ? bnAmount : bn(1),
```

Comment:
According to the changelog [message](https://github.com/FuelLabs/fuels-ts/pull/1603/files#diff-7e2912a79a5d3caf3f2ac8e34f4decd617c7805f6a4e01cdaa41749a284cf6e4) and your [test](https://github.com/FuelLabs/fuels-ts/pull/1603/files#diff-f755dbc4776c2abee9a5e41e77e122fbf3eecaf6aa5dec65e211fee5b1858a53R113), this should be like this:

```suggestion
    amount: bnAmount.toNumber() < 1 ? bn(1) : bnAmount,
```

Reply:
In the [test](https://github.com/FuelLabs/fuels-ts/pull/1603/files#diff-f755dbc4776c2abee9a5e41e77e122fbf3eecaf6aa5dec65e211fee5b1858a53R113), remember to assert it became `1` instead of a generic [`truthy`](https://github.com/FuelLabs/fuels-ts/pull/1603/files#diff-f755dbc4776c2abee9a5e41e77e122fbf3eecaf6aa5dec65e211fee5b1858a53R117) value.

Reply:
Related to the test, the amount of a given resource is determined by the chain config of the node so I can't assert a certain value, just that it's greater than or equal to 1.

Reply:
You can validate if `coinQuantityfy()` explicitly turns `amount: 0` into `amount: 1`.

This will test your code addition without margin for false positives.

Reply:
https://github.com/FuelLabs/fuels-ts/pull/1603/commits/1c44eeca2e5e41a599d3c34d1d4659d849063591